python/cudf/cudf/core/join/join.py

```diff
@@ -339,6 +347,21 @@ def _merge_results(self, left_result: Frame, right_result: Frame) -> Frame:
 
         return result
 
+    @property
+    def _result_has_multiindex_columns(self):
```

Comment:
Since this can be determined from the merge configuration I might suggest just computing it on init and storing it as an attribute, or as part of the merge parameter preprocessing step